### PR TITLE
feat: integrated star printer functionality

### PR DESCRIPTION
**Pull request by GaneshTILLX**
## Description

Added interface with star printers for use.
## Related Issues

Issue: #XXX

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Screenshots (if appropriate)

### CODE REVIEW

File: example/src/App.tsx [hunk 1]
```diff
@@ -25,22 +20,21 @@ export default function App() {
...
-          printer.port,
-          buffer,
-          PrinterBrand.EPSON,
-          PrinterSeries.TM_M30
-        );
-      } catch (err) {
-        console.log('error', err);
-      }
+    // for (let i = 0; i < 100; i++) {
+    let spont: number = Math.floor(Math.random() * PRINTERS.length);
```
Comment (mdsiddiq-till) [inline example/src/App.tsx:24]: nit: what is meant by spont?

File: src/star.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    let printer = new StarPrinter(settings);
+    let callback = (callbackFunc: (val: any) => void, val: any) => () =>
+      callbackFunc(val);
+    let lazyCallback = () => errorCallback('Failed to Print');
+
+    try {
+      await printer.open();
+      const status = await printer.getStatus();
+      console.log('status ' + JSON.stringify(status));
+      await printer.printRawData([...text]);
```
Comment (mdsiddiq-till) [inline src/star.ts:34]: await printer.printRawData(text);
doesn't work?

File: src/star.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    );
+
+    let printer = new StarPrinter(settings);
+    let callback = (callbackFunc: (val: any) => void, val: any) => () =>
+      callbackFunc(val);
+    let lazyCallback = () => errorCallback('Failed to Print');
+
+    try {
+      await printer.open();
+      const status = await printer.getStatus();
```
Comment (mdsiddiq-till) [inline src/star.ts:32]: Query:
This is just for logging status right? we don't do anything based on this.

File: src/star.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      console.log('status ' + JSON.stringify(status));
+      await printer.printRawData([...text]);
+      lazyCallback = callback(successCallback, 'Successfully printed');
+    } catch (err) {
+      lazyCallback = callback(errorCallback, err);
+    } finally {
+      await printer.close();
+      await printer.dispose();
+      lazyCallback();
```
Comment (mdsiddiq-till) [inline src/star.ts:41]: The way of using lazyCallback is interesting. 👍

File: tsconfig.json [hunk 5]
```diff
@@ -17,8 +17,8 @@
     "noImplicitReturns": true,
     "noImplicitUseStrict": false,
     "noStrictGenericChecks": false,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
+    "noUnusedLocals": false,
```
Comment (mdsiddiq-till) [inline tsconfig.json:20]: are these changes needed for the above code?

File: example/ios/ThermalReceiptPrinterExample/Info.plist [hunk 6]
```diff
@@ -2,6 +2,10 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>UISupportedExternalAccessoryProtocols</key>
```
Comment (mdsiddiq-till) [inline example/ios/ThermalReceiptPrinterExample/Info.plist:5]: Query:
What is about https://github.com/star-micronics/react-native-star-io10#4-apple-approval-process-for-star-mfi-applications?

It is said int he documentation, that for bluetooth this is needed

File: example/src/App.tsx [hunk 7]
```diff
@@ -13,8 +9,7 @@ interface Printer {
   port: number;
 }
 const PRINTERS: Array<Printer> = [
```
Comment (mdsiddiq-till) [inline example/src/App.tsx:11]: nit: We could have put other printers also here.

File: ios/RNNetPrinter.m [hunk 8]
```diff
@@ -28,28 +28,29 @@ - (dispatch_queue_t)methodQueue
                   success:(RCTResponseSenderBlock)successCallback
                   fail:(RCTResponseSenderBlock)errorCallback) {
     @try {
-        int number;
-        if ([series isEqual:@"TM_M30"]){
+        if ([brand  isEqual: @"EPSON"]) {
```
Comment (mdsiddiq-till) [inline ios/RNNetPrinter.m:31]: nit: _Looks like indentation is disturbed in this file._

**Review by mdsiddiq-till — COMMENTED** [review]
👍

File: example/ios/ThermalReceiptPrinterExample/Info.plist (hunk 6 above)
Comment (GaneshTILLX) [reply example/ios/ThermalReceiptPrinterExample/Info.plist:5]: We will need to submit this before releasing to the App Store
